src/Orleans/Messaging/Message.cs

```diff
@@ -440,24 +444,15 @@ public Message()
             headerBytes = null;
         }
 
-        private Message(Categories type, Directions subtype)
-            : this()
-        {
-            Category = type;
-            Direction = subtype;
-        }
-
         internal static Message CreateMessage(InvokeMethodRequest request, InvokeMethodOptions options)
         {
-            var message = new Message(
```

Comment:
But there's another problem - every object (at least message have to be disposed only once, otherwise object can be returned back to the pool in the middle of it's life. So I would also introduce throwing exceptions on double disposing, that will be thrown eventually even in aforementioned case..